packages/dds/tree/src/shared-tree/treeAlpha.ts

```diff
@@ -339,6 +353,44 @@ export interface TreeAlpha {
 	 * Otherwise, this returns the key of the field that it is under (a `string`).
 	 */
 	key2(node: TreeNode): string | number | undefined;
+
+	/**
+	 * Gets the child of the given node with the given property key if a child exists under that key.
+	 *
+	 * @remarks Unknown optional fields of Object nodes cannot be accessed using this method.
+	 *
+	 * @param node - The parent node whose child is being requested.
+	 * @param key - The property key under the node under which the child is being requested.
+	 * Note: this is the "property key", not the "stored key".
```

Comment:
Clarify that keys for array nodes are just the numeric index and that's why it can be a number?